yb-voyager/cmd/importData.go

```diff
@@ -355,15 +353,38 @@ func importData() {
 		fmt.Printf("Number of parallel imports jobs at a time: %d\n", parallelism)
 	}
 
-	if parallelism > SPLIT_FILE_CHANNEL_SIZE {
-		splitFileChannelSize = parallelism + 1
+	determineTablesToImport()
+	if startClean {
+		cleanImportState()
+	}
+	if target.VerboseMode {
+		fmt.Printf("all the tables to be imported: %v\n", allTables)
+		fmt.Printf("tables left to import: %v\n", importTables)
 	}
-	splitFilesChannel := make(chan *SplitFileImportTask, splitFileChannelSize)
 
-	generateSmallerSplits(splitFilesChannel)
-	go doImport(splitFilesChannel, parallelism, connPool)
-	checkForDone()
-	time.Sleep(time.Second * 2)
+	if len(importTables) == 0 {
+		fmt.Printf("All the tables are already imported, nothing left to import\n")
+	} else {
+		fmt.Printf("Importing tables: %v\n", importTables)
+		initializeImportDataStatus(exportDir, importTables)
+		if !disablePb {
+			go importDataStatus()
+		}
+		poolSize := 20
+		if parallelism > poolSize {
```

Comment:
Rather i will suggest to have: `poolSize = max(20, parallelism * 2)`

since we expect all parallel jobs to start and end at the same time, I expect there are chances that half of the parallel jobs after the first iteration might stay always idle.

Reply:
I have changed it to:
```
poolSize = parallelism + parallelism/2
```
We should not go overboard with the `parallelism*2`. With higher values of parallelism this `*2` will result in large number of extra batches being created which can lead to batch data being flushed to the disk (today contents written to the batch files almost never hit the disk).